pkg/shared/globalconfig.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 BlueData Software, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	kdv1 "github.com/bluek8s/kubedirector/pkg/apis/kubedirector.bluedata.io/v1alpha1"
+	"sync"
+)
+
+var (
+	globalConfig     *kdv1.KubeDirectorConfig
+	globalConfigLock sync.RWMutex
+)
+
+// getNativeSystemdSupport extracts the flag definition from the
+// globalConfig CR data if present, otherwise returns false
+func GetNativeSystemdSupport() bool {
```

Comment:
Seems like these read-accessor functions should also lock the globalConfigLock. I don't think that pointer-variable update is guaranteed to be atomic in the Go spec.